karoo_gp/base_class.py

```diff
@@ -1787,1535 +2043,1588 @@ class labels provided in the data .csv. Outputs an array of tuples
-                         display=['db'])
-                self.pause(display=['db'])
-
-                # insert new 'branch' at point of mutation 'branch_top'
-                # in tourn_winner 'tree'
-                tree = self.fx_evolve_branch_insert(tree, branch)
-                # because we already know the maximum depth to which this
-                # branch can grow, there is no need to prune after insertion
-
-        tree = self.fx_evolve_fitness_wipe(tree)  # wipe fitness data
-
-        return tree
-
-
-    def fx_evolve_crossover(self, parent, branch_x, offspring, branch_y):
-
-        '''
-        Refer to the method fx_nextgen_crossover() for a full description
-        of the genetic operator Crossover.
-
-        This method is called twice to produce 2 offspring per pair of
-        parent Trees. Note that in the method 'karoo_fx_crossover' the
-        parent/branch relationships are swapped from the first run to
-        the second, such that this method receives swapped components to
-        produce the alternative offspring. Therefore 'parent_b' is first
-        passed to 'offspring' which will receive 'branch_a'. With the
-        second run, 'parent_a' is passed to 'offspring' which will receive
-        'branch_b'.
-
-        Called by: fx_nextgen_crossover
-
-        Arguments required: parent, branch_x, offspring,
-            branch_y (parents_a / _b, branch_a / _b from fx_nextgen_crossover()
-        '''
-
-        # pointer to the top of the 1st parent branch passed from fx_nextgen_crossover()
-        crossover = int(branch_x[0])
-        # pointer to the top of the 2nd parent branch passed from fx_nextgen_crossover()
-        branch_top = int(branch_y[0])
-
-        self.log('\n\n\033[33m *** Crossover *** \033[0;0m', display=['db'])
-
-        # if the branch from the parent contains only one node (terminal)
-        if len(branch_x) == 1:
-
-            self.log(f'\t\033[36m  terminal crossover from \033[1mparent '
-                     f'{parent[0][1]} \033[0;0m\033[36mto \033[1moffspring '
-                     f'{offspring[0][1]} \033[0;0m\033[36mat node\033[1m '
-                     f'{branch_top} \033[0;0m', display=['i'])
-
-            self.log(f'\n\033[36m In a copy of one parent:\033[0;0m\n {offspring}',
-                     display=['db'])
-            self.log(f'\n\033[36m ... we remove nodes\033[1m {branch_y} '
-                     f'\033[0;0m\033[36mand replace node\033[1m {branch_top} '
-                     '\033[0;0m\033[36mwith a terminal from branch_x\033[0;0m',
-                     display=['db'])
-            self.pause(display=['db'])
-
-            offspring[5][branch_top] = 'term'  # replace type
-            # replace label with that of a particular node in 'branch_x'
-            offspring[6][branch_top] = parent[6][crossover]
-            offspring[8][branch_top] = 0  # set terminal arity
+            rnd = rng.integers(0, len(terminals) - 1)
```

Comment:
As before, check the `-1`.
```suggestion
            rnd = rng.integers(len(terminals) - 1)
```